Source/Particles/WarpXParticleContainer.cpp

```diff
@@ -1143,10 +1156,10 @@ WarpXParticleContainer::ApplyBoundaryConditions (ParticleBoundaries& boundary_co
 #endif
                                                               z, zmin, zmax,
                                                               ux[i], uy[i], uz[i], particle_lost,
-                                                              boundary_conditions);
+                                                              this->m_boundary_conditions, engine);
```

Comment:
You can't use `this` inside a GPU kernel.  Given the prior code, it should be valid to copy `auto boundary_conditions = m_boundary_conditions;` outside the `ParallelFor`, and then reference that local copy.

Reply:
That was my thinking too but if I do that I get an error:
```
./Source/Particles/WarpXParticleContainer.cpp: In member function ‘void WarpXParticleContainer::ApplyBoundaryConditions()’:
./Source/Particles/WarpXParticleContainer.cpp:1120:32: error: use of deleted function ‘ParticleBoundaries::ParticleBoundaries(const ParticleBoundaries&)’
 1120 |     auto boundary_conditions = m_boundary_conditions;
```